Common/cpp/Tools/BackgroundQueue.cpp

```diff
@@ -0,0 +1,97 @@
+#include "BackgroundQueue.h"
+
+#include <utility>
+
+namespace reanimated {
+
+BackgroundQueue::BackgroundQueue(const std::string &name) : name_(name) {
+  thread_ = std::thread([this] {
+#if __APPLE__
+    pthread_setname_np(name_.c_str());
+#endif
+    runLoop();
+  });
+#ifdef ANDROID
+  pthread_setname_np(thread_.native_handle(), name_.c_str());
+#endif
+}
+
+BackgroundQueue::~BackgroundQueue() {
+  running_ = false;
+  cv_.notify_all();
+  thread_.join();
```

Comment:
Why `join` instead of `detach`?

Reply:
That's an excellent question, I've copied this code from somewhere else. We'll need to decide if we want to change it or not. cc @kmagiera

Reply:
Changed to `thread_.detach()`. Background queue holds a queue of jobs (`std::function`) which retain worklet runtime and worklet (as shared pointers).